internal/database/txn/transaction.go

```diff
@@ -244,6 +259,20 @@ func (t *RetryingTxnRunner) run(ctx context.Context, fn func(context.Context) er
 		return errors.Trace(err)
 	}
 
+	// Txn ID is used to track the whole transaction, from the start to the end.
+	// This also includes the BEGIN, COMMIT and ROLLBACK.
+	// As these are processed in real-time, it will require the user of the
+	// logging infrastructure to be able to track the txnID and to patch the
+	// log lines together. The alternative is to use the tracing infrastructure
+	// which will give it a context, but that's a lot of overhead to run in
+	// production for every query.
+	txnID := atomic.AddUint64(&t.txnID, 1)
```

Comment:
Yeah, this is a known limitation. Most transactions will be far enough apart if we bounce and start from 0. Worst case it will be `0, 1, 2, 3, 4, 5, 0, 1, 2, 3, 4, 5` (although it will likely be in the hundreds)...